### ISSUE TITLE
Training loss =Nan for Pascal VOC training after 12000 iterations

### ISSUE THREAD

**Issue by vj-1988** [weiliu89/caffe#93]
I am getting nan as trainig loss after 12000 iterations for Pascal VOC training. I followed the steps given in home page. The training loss started at 14 and was gradually decreasing until it suddenly went to Nan. Did anyone face similar issue?

**Comment by Li1991** [comment]
I have encountered this problem, have you solved it? Can you tell me how to solve it?

**Comment by shawn-tian** [comment]
maybe you can try to decrease the learning rate

**Comment by weiliu89** [comment]
@vj-1988 @Li1991 It might due to a bug and hopefully has been fixed in e4cde71
What is the batch size you use during training? If it is small, then it is likely that there is no ground truth in a batch of images, and thus cannot find any match, which makes the loss goes up very high suddenly. I usually used 8 images in a batch, and the possibility of having such bug is very very small.

You could pull the latest code and try again...

**Comment by divamgupta** [comment]
Hi @weiliu89 ,

I am using the latest code. But the loss comes to nan if I use a batch size of 8. What should I do if a have a small GPU?

**Comment by weiliu89** [comment]
@divamgupta Did you set `accum_batch` to 32? Another possible way is to reduce the learning rate a bit if your batch size is small.

**Comment by vj-1988** [comment]
@divamgupta : The author @weiliu89 is correct. I first tried training the VOC with only one GTX960. My batch size was very small and hence I faced this issue. Later, I got two Titanx cards with which I did not face this issue since I was able to start the training with default training parameters mentioned in ssd_pascal.py. 

Regarding the accum_batch , the nan did not appear again in my gtx960 card, but I remember the final accuracy was lesser.
